### PR TITLE
Improve release docs

### PR DESCRIPTION
**Pull request by aanand**
Incorporating @mnowster's [gist feedback](https://gist.github.com/aanand/e567bd8d6a5d8e28c829#gistcomment-1550971).

ping @moxiegirl

### CODE REVIEW

**Comment by mnowster** [comment]
Yay! Thanks for the clarification @aanand :crystal_ball:

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -78,9 +84,29 @@ Building a Compose release
 
         git push git@github.com:docker/compose.git $TAG
 
-8.  Create a release from the tag on GitHub.
+8.  Draft a release from the tag on GitHub.
+
+    - Go to https://github.com/docker/compose/releases and click "Draft a new release".
+    - In the "Tag version" dropdown, select the tag you just pushed.
+
```
Comment (moxiegirl) [inline RELEASE_PROCESS.md]: Can you show an actual example for your template?  What follows after 9 looks like sub procedure, if it is, use step-wise numbers (not Firstly/Secondly etc) as they are easier to follow.

Comment (aanand) [inline RELEASE_PROCESS.md]: @moxiegirl It's not sub-procedure, it's copy for the release notes which is meant to be copy/pasted into the GitHub release. What would be the best way to make that clear?

Comment (moxiegirl) [inline RELEASE_PROCESS.md]: Ah, @aanand ok...I'm sorted it was clarified by doing a *VIEW* for the format version.  Apologies LGTM

**Comment by mnowster** [comment]
LGTM

**Pull request merged**